sarek blog post (#2472)
nf-core/website: src/content/blog/2024/sarek_paper.mdx

---
title: 'nf-core/sarek paper'
subtitle: 'Harder, Better, Faster, Stronger'
pubDate: 2024-04-25T00:13:00+01:00

headerImage: '/assets/images/blog/sarek/skierfe.jpg'

> headerImageAlt: 'Skierfe'

Reviewer: That's not a good description for people not seeing the image 😉

authors:
    - 'FriederikeHanssen'
    - 'maxulysse'
label:

>     - 'community'

Reviewer: ```suggestion
    - 'pipeline'
```

import sarek_subway from '@assets/images/blog/sarek/sarek_subway.png';
import cram_meme from '@assets/images/blog/sarek/cram_meme.jpg';
import cram_graph from '@assets/images/blog/sarek/storage_preprocessing_reduced_cumulative.png';
import dataflow from '@assets/images/blog/sarek/paper_dataflow.png';
import illumina from '@assets/images/blog/sarek/benchmarking_snvs_combination.png';
import not_illumina from '@assets/images/blog/sarek/benchmarking_snps_combination_nonillumina.png';

import { Image } from 'astro:assets';

We are extremely happy to see this [paper](https://academic.oup.com/nargab/article/6/2/lqae031/7658070) out describing the changes and
updates to nf-core/sarek, the DNA variant calling pipeline, in the last several years.

In 2020 we embarked on the journey of rewriting the whole pipeline in DSL2. One of the major motivations was to bring down
cloud computing costs and generally reduce storage space and computational resources. Overall, we reduced computational
costs on AWS (last summer, using spot instances) by 70% to about $20 from FASTQs to annotated VCFs using Strelka, Manta, and VEP.

> But first an overview - no nf-core pipeline without a metro map :

Reviewer: ```suggestion
But first an overview - no nf-core pipeline without a metro map:
```

---
title: 'nf-core/sarek paper'
subtitle: 'Harder, Better, Faster, Stronger'

> pubDate: 2024-04-26T00:13:00+01:00

Reviewer 2: shouldn't the date be today's?

Author: yeah but when I put date today it put april 24ths on the preview 🤨

---
title: 'nf-core/sarek paper'
subtitle: 'Harder, Better, Faster, Stronger'
pubDate: 2024-04-25T00:13:00+01:00

headerImage: '/assets/images/blog/sarek/skierfe.jpg'

> headerImageAlt: 'Skierfe'

Reviewer 2: ```suggestion
headerImageAlt: 'Mount Skierfe outlined showing the inspiration for the Sarek logo'
```

import not_illumina from '@assets/images/blog/sarek/benchmarking_snps_combination_nonillumina.png';

import { Image } from 'astro:assets';

We are extremely happy to see this [paper](https://academic.oup.com/nargab/article/6/2/lqae031/7658070) out describing the changes and
updates to nf-core/sarek, the DNA variant calling pipeline, in the last several years.

In 2020 we embarked on the journey of rewriting the whole pipeline in DSL2. One of the major motivations was to bring down
cloud computing costs and generally reduce storage space and computational resources. Overall, we reduced computational
costs on AWS (last summer, using spot instances) by 70% to about $20 from FASTQs to annotated VCFs using Strelka, Manta, and VEP.

But first an overview - no nf-core pipeline without a metro map :

<Image src={sarek_subway} alt="Metromap of nf-core/sarek" height="300" />

We ditched BAM format where possible and switched to CRAM saving us 4x work storage space.

<Image src={cram_meme} alt="Toy story meme" height="300" />
<Image src={cram_graph} alt="Bar chart showing reduction in storage space usage" height="300" />

> The `s`plitFastq()` operator was replaced by a FastP process to split the Fastq

Reviewer 2: ```suggestion
The `splitFastq()` operator was replaced by a FastP process to split the Fastq
```

> files before alignment (default 12), and changed the default grouping of the intervals for BQSR to 21 reducing storage space another 4x

Reviewer 2: ```suggestion
files before alignment (default 12) plus replacing trimgalore! for read trimming, and changed the default grouping of the intervals for BQSR to 21 reducing storage space another 4x
```
